security/subsystem/src/main/java/org/jboss/as/security/elytron/ElytronRealmResourceDefinition.java

```diff
@@ -0,0 +1,134 @@
+ *  published by the Free Software Foundation; either version 2.1 of
+ *  the License, or (at your option) any later version.
+ *
+ *  This software is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ *  Lesser General Public License for more details.
+ *
+ *  You should have received a copy of the GNU Lesser General Public
+ *  License along with this software; if not, write to the Free
+ *  Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ *  02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+
+package org.jboss.as.security.elytron;
+
+import org.jboss.as.controller.AbstractAddStepHandler;
+import org.jboss.as.controller.AttributeDefinition;
+import org.jboss.as.controller.OperationContext;
+import org.jboss.as.controller.OperationFailedException;
+import org.jboss.as.controller.PathAddress;
+import org.jboss.as.controller.PathElement;
+import org.jboss.as.controller.RestartParentWriteAttributeHandler;
+import org.jboss.as.controller.ServiceRemoveStepHandler;
+import org.jboss.as.controller.SimpleAttributeDefinition;
+import org.jboss.as.controller.SimpleAttributeDefinitionBuilder;
+import org.jboss.as.controller.SimpleResourceDefinition;
+import org.jboss.as.controller.capability.RuntimeCapability;
+import org.jboss.as.controller.descriptions.StandardResourceDescriptionResolver;
+import org.jboss.as.controller.registry.AttributeAccess;
+import org.jboss.as.controller.registry.ManagementResourceRegistration;
+import org.jboss.as.controller.registry.OperationEntry;
+import org.jboss.as.security.Constants;
+import org.jboss.as.security.SecurityExtension;
+import org.jboss.as.security.plugins.SecurityDomainContext;
+import org.jboss.as.security.service.SecurityDomainService;
+import org.jboss.dmr.ModelNode;
+import org.jboss.dmr.ModelType;
+import org.jboss.msc.service.ServiceController;
+import org.jboss.msc.service.ServiceName;
+import org.wildfly.security.auth.server.SecurityRealm;
+
+/**
+ * Defines a resource that represents an elytron-compatible realm that will be exported by the security subsystem.
+ *
+ * @author <a href="mailto:sguilhen@redhat.com">Stefan Guilhen</a>
+ */
+public class ElytronRealmResourceDefinition extends SimpleResourceDefinition {
+
+    public static final PathElement ELYTRON_REALM_PATH = PathElement.pathElement(Constants.ELYTRON_REALM);
+
+    static final String SECURITY_REALM_CAPABILITY = "org.wildfly.security.security-realm";
+
+    static final RuntimeCapability<Void> SECURITY_REALM_RUNTIME_CAPABILITY = RuntimeCapability
+            .Builder.of(SECURITY_REALM_CAPABILITY, true, SecurityRealm.class)
+            .build();
+
+    static final SimpleAttributeDefinition DELEGATE_SECURITY_DOMAIN =
+            new SimpleAttributeDefinitionBuilder(Constants.DELEGATE_SECURITY_DOMAIN, ModelType.STRING, false)
```

Comment:
Definitely. If I'm not mistaken, this is in the roadmap for the security subsystem (according to that capabilities doc project) and I didn't add it in this commit because I was not sure it was not being already done upstream by somebody else.

The best approach seems to handle this capability upstream first - it will have to be present for WF 10 anyway and once it is merged upstream we can merge the relevant commits into our branch and I can then change this attribute so it references the domain capability.